tensorflow/python/kernel_tests/xent_op_test.py

```diff
@@ -171,28 +171,41 @@ def testGradient(self):
       x = nn_ops.softmax_cross_entropy_with_logits(labels=l, logits=f,
                                                    name="xent")
       err = gradient_checker.compute_gradient_error(f, [3, 4], x, [3])
+
+      # Check that no extra computation performed. When only first derivative is requested,
+      # second derivative must not be computed. So when there is no second derivative,
+      # there is no `BatchMatMul` op in the graph.
+      op_names = [op.op_def.name for op in sess.graph.get_operations() if op.op_def]
+      self.assertNotIn('BatchMatMul', op_names)
+
     print("cross entropy gradient err = ", err)
     self.assertLess(err, 5e-8)
 
   def testSecondGradient(self):
-    with self.test_session():
-      l = constant_op.constant([0.0, 0.0, 1.0, 0.0,
-                                1.0, 0.0, 0.0, 0.0,
-                                0.0, 0.5, 0.0, 0.5], shape=[12],
+    with self.test_session() as sess:
+      l = constant_op.constant([0.0, 0.0, 1.0/3, 0.0,
+                                1.0/3, 0.0, 0.0, 0.0,
+                                0.0, 0.5/3, 0.0, 0.5/3], shape=[12],
                                dtype=dtypes.float64, name="l")
       f = constant_op.constant([0.1, 0.2, 0.3, 0.4,
                                 0.1, 0.4, 0.9, 1.6,
                                 0.1, 0.8, 2.7, 6.4], shape=[12],
                                dtype=dtypes.float64, name="f")
       x = nn_ops.softmax_cross_entropy_with_logits(labels=l, logits=f,
                                                    name="xent")
-      loss = math_ops.reduce_mean(x)
+      loss = math_ops.reduce_sum(x)
+
+      trivia_loss = -math_ops.reduce_sum(l * math_ops.log(nn_ops.softmax(f)))
+
+      hessians = sess.run(gradients_impl.hessians(loss, [f]))
+      trivia_hessians = sess.run(gradients_impl.hessians(trivia_loss, [f]))
+
+      self.assertAllClose(hessians, trivia_hessians)
```

Comment:
Use `gradient_checker.compute_gradient_error` instead of writing out the gradient manually a second time.